### DISCUSSION TITLE
List of network choices doesn't filter based on selected protocol network

### DISCUSSION THREAD

**Discussion by alex-pakalniskis** [graphprotocol/graph-tooling#1628] [Ideas]
## Problem
When creating a new subgraph using `graph init`, a user selects the desired protocol network from which to index data.

![Graph Init protocol selection](https://github.com/graphprotocol/graph-tooling/assets/43630382/688ed913-af8b-4035-bf42-fce9ca523c6f)

After selecting the protocol network to index, users are presented with the global list of sub-choices that aren't filtered to the specific protocol network chosen previously. 

![NEAR example](https://github.com/graphprotocol/graph-tooling/assets/43630382/803b5e56-0b76-42d9-b747-bb2af1855dad)

![Cosmos example](https://github.com/graphprotocol/graph-tooling/assets/43630382/f5562459-17af-42aa-bb59-bb3ed313031f)

This is potentially confusing for newer users as "mainnet" could refer to Ethereum mainnet or NEAR/Cosmos mainnet in above figures.

## Solution

Filter sub-choices based on selected protocol network, i.e. only show NEAR relevant choices if NEAR chosen, etc.

## Clarification
Is the current behavior there for a specific reason? (i.e. some technical requirement)

**Comment by alex-pakalniskis** [comment]
@saihaj thoughts?

**Reply by saihaj** [reply]
> Is the current behavior there for a specific reason? (i.e. some technical requirement)

the way we are getting these lists are just static. Need to figure out how to structure these better so we can filter out

**Comment by YaroShkvorets** [comment]
solved with registry
